src/der.rs

```diff
@@ -104,14 +104,23 @@ pub(crate) fn expect_tag<'a>(
 pub(crate) fn read_tag_and_get_value<'a>(
     input: &mut untrusted::Reader<'a>,
 ) -> Result<(u8, untrusted::Input<'a>), Error> {
-    ring::io::der::read_tag_and_get_value(input).map_err(|_| Error::BadDer)
+    read_tag_and_get_value_limited(input, THREE_BYTE_DER_LEN_SIZE_LIMIT)
 }
 
+// Short-form DER encoded lengths of three bytes can express lengths up to the following limit.
+// The first byte provides 7 usable bits (losing the MSB to encoding overhead) while the second and
+// third bytes provide 8 bits each. 7 + 8 + 8 = 23, making the maximum length 2^23 - 1.
+//
+// The upstream ring::io::der::read_tag_and_get_value() function limits itself to up to three byte
+// short-form DER lengths, and so this limit represents the maximum length that was possible to
+// read before the introduction of the read_tag_and_get_value_limited function.
+#[allow(clippy::as_conversions)] // infallible known good usize.
+pub(crate) const THREE_BYTE_DER_LEN_SIZE_LIMIT: usize = (2_i32.pow(23) - 1) as usize;
```

Comment:
This name seems a little too onerous? I'd probably strip it to just `THREE_BYTE_DER_SIZE`.

Reply:
I reworded the `crate` exported constant name to match your suggestion. It's an export of a wordier constant `LONG_FORM_LEN_TWO_BYTES` that better matches a series of constants internal to the der module I added to help remove magic numbers from the parse logic. 

If having two constants for the same value (one with a short name and crate visibility, and one with a longer name and private visibility) isn't to your taste I can revisit.